cli/command/system/prune.go

```diff
@@ -67,6 +67,14 @@ func runBuildCachePrune(dockerCli command.Cli, _ opts.FilterOpt) (uint64, string
 }
 
 func runPrune(dockerCli command.Cli, options pruneOptions) error {
+
+	progressChan := pruneProgressChan{
+		errorChan:         make(chan error),
+		isEventsListening: make(chan bool),
+	}
+
+	runPruneProgress(dockerCli, &progressChan, options)
```

Comment:
I think this should be after the 3 checks below, so that 1) related code stays together, and 2) setting `options.pruneBuildCache = false` is read correctly (it isn't now).

Once it's moved to immediately before the select, I wonder if the `progressChan` could be moved into `runPruneProgress()` since it seems nothing else uses it:

```go
if err := startPruneProgress(dockerCli, options); err != nil {
    return err
}

return runAllPrunes(dockerCli, options)
```